plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/thrift/TokenFetchingMetastoreClientFactory.java

```diff
@@ -106,12 +119,22 @@ private DelegationToken getDelegationToken(String username)
 
     private DelegationToken loadDelegationToken(String username)
     {
-        try (ThriftMetastoreClient client = createMetastoreClient()) {
-            return new DelegationToken(System.nanoTime(), client.getDelegationToken(username));
+        try {
+            // added retry and stats for the thrift delegation token
+            return retry()
+                    .stopOnIllegalExceptions()
+                    .run("getDelegationToken", stats.getThriftDelegationToken().wrap(() -> {
+                        try (ThriftMetastoreClient client = createMetastoreClient()) {
+                            return new DelegationToken(System.nanoTime(), client.getDelegationToken(username));
+                        }
+                    }));
         }
         catch (TException e) {
             throw new TrinoException(HIVE_METASTORE_ERROR, e);
         }
+        catch (Exception e) {
+            throw propagate(e);
```

Comment:
We can inline this method as it is only used in one place

Reply:
updated